content/en/blog/_posts/2023-01-17-confidential-kubernetes.md

```diff
@@ -0,0 +1,346 @@
+API adaptor that starts pod VMs in the cloud. CoCo can also be run on
+bare metal, providing strong isolation even in multi-tenant environments.
+
+### Managed confidential Kubernetes
+
+[Azure](https://learn.microsoft.com/en-us/azure/confidential-computing/confidential-node-pool-aks) and
+[GCP](https://cloud.google.com/blog/products/identity-security/announcing-general-availability-of-confidential-gke-nodes)
+both support the use of confidential virtual machines as worker nodes for their managed Kubernetes offerings.
+
+Both services aim for better workload protection and security guarantees by enabling memory encryption
+for container workloads. However, they don't seek to fully isolate the cluster or workloads against
+the service provider or infrastructure. Specifically, they don't offer a dedicated confidential control
+plane or expose attestation capabilities for the confidential cluster/nodes.
+
+Azure also enables
+[Confidential Containers](https://learn.microsoft.com/en-us/azure/confidential-computing/confidential-nodes-aks-overview)
+in their managed Kubernetes offering. They support the creation based on
+[Intel SGX enclaves](https://learn.microsoft.com/en-us/azure/confidential-computing/confidential-containers-enclaves)
+and [AMD SEV-based VMs](https://techcommunity.microsoft.com/t5/azure-confidential-computing/microsoft-introduces-preview-of-confidential-containers-on-azure/ba-p/3410394).
+
+### Constellation
+
+[Constellation](https://github.com/edgelesssys/constellation) is a Kubernetes engine that aims to
+provide the best possible data security. Constellation wraps your entire Kubernetes cluster into
+a single confidential context that is shielded from the underlying cloud infrastructure. Everything
+inside is always encrypted, including at runtime in memory. It shields both the worker and control
+plane nodes. In addition, it already integrates with popular CNCF software such as Cilium for
+secure networking and provides extended CSI drivers to write data securely.
+
+### Occlum and Gramine
+
+[Occlum](https://occlum.io/) and [Gramine](https://gramineproject.io/) are examples of open source
+library OS projects that can be used to run unmodified applications in SGX enclaves. They
+are member projects under the CCC, but similar projects and products maintained by companies
+also exist. With these libOS projects, existing containerized applications can be
+easily converted into confidential computing enabled containers. Many curated prebuilt
+containers are also available.
+
+## Where are we today? Vendors, limitations, and FOSS landscape
+
+As we have seen in the previous sections, Confidential Computing is a powerful new concept
+to improve security, but we are still in the (early) adoption phase. New products are
+starting to emerge to take advantage of the unique properties.
+
+Google and Microsoft are the first major cloud providers to have confidential offerings.
```

Comment:
```suggestion
Google and Microsoft are the first major cloud providers to have confidential offerings that
can run unmodified applications inside a protected boundary.
```
Nitro enclaves _are_ a thing (and have been since 2020ish), but not for unmodified workloads.